nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -892,16 +988,16 @@ in
               ACTIVE_RECORD_ENCRYPTION_DETERMINISTIC_KEY="$(cat ${cfg.activeRecordEncryptionDeterministicKeyFile})"
               ACTIVE_RECORD_ENCRYPTION_KEY_DERIVATION_SALT="$(cat ${cfg.activeRecordEncryptionKeyDerivationSaltFile})"
               ACTIVE_RECORD_ENCRYPTION_PRIMARY_KEY="$(cat ${cfg.activeRecordEncryptionPrimaryKeyFile})"
-              SECRET_KEY_BASE="$(cat ${cfg.secretKeyBaseFile})"
+              SECRET_KEY_BASE="$(cat ${cfg.secrets.SECRET_KEY_BASE})"
               VAPID_PRIVATE_KEY="$(cat ${cfg.vapidPrivateKeyFile})"
               VAPID_PUBLIC_KEY="$(cat ${cfg.vapidPublicKeyFile})"
             ''
+            + (lib.concatMapAttrsStringSep "" (name: path: ''
+              ${name}="$(cat '${path}')"
```

Comment:
I just tried to set up translation with libretranslate through the LIBRE_TRANSLATE_API_KEY env and it took me a while to realize that secrets is not generic. I think we should add another generic block like we have for settings for secrets. 😅

Reply:
Oh makes sense. I added eou pipefail for the script. Thanks for testing the feature.